TLS_E2E_Automation_Framework/cypress/e2e/tls_page_elements/loginLogout.js

```diff
@@ -0,0 +1,194 @@
+    });
+  cy.log("The user is successfully logged out from homepage");
+
+  // Navigate to an article page and verify the user is logged out
+  cy.get(articleLink).eq(0).click();
+  cy.get(loginButton)
+    .invoke("text")
+    .then((text) => {
+      expect(text).to.equal("Login");
+    });
+  cy.log("The user is successfully logged out from article page");
+}
+
+// Validate login functionality from the homepage
+export function validateLoginFromHomepage() {
+  cy.log("Verify the user tries to login from homepage");
+
+  // Visit the homepage URL and accept the cookie banner
+  cy.visit(url, { timeout: 20000 });
+  cy.acceptCookieBanner();
+
+  // Verify the user is currently logged out
+  cy.get(subscribeButton)
+    .invoke("text")
+    .then((text) => {
+      expect(text).to.equal("Subscribe");
+    });
+  cy.log("The user is logged out");
+
+  // Perform login and verify
+  performLogin();
+  verifyLogin();
+}
+
+// Validate login functionality from an article page
+export function validateLoginFromArticlePage() {
+  cy.log("Verify the user tries to login from article page");
+
+  // Visit the homepage URL and accept the cookie banner
+  cy.visit(url, { timeout: 20000 });
+  cy.acceptCookieBanner();
+
+  // Navigate to the first article and verify the user is logged out
+  cy.get(articleLink).eq(0).click();
+  cy.get(subscribeButton)
+    .invoke("text")
+    .then((text) => {
+      expect(text).to.equal("Subscribe");
+    });
+  cy.log("The user is logged out");
+
+  // Perform login and verify
+  performLogin();
+  verifyLogin();
+}
+
+// Validate logout functionality from the homepage
+export function validateLogoutFromHomepage() {
+  cy.log("Verify the logout scenario from the homepage");
+
+  // Visit the homepage URL, accept the cookie banner, and perform login
+  cy.visit(url, { timeout: 20000 });
+  cy.acceptCookieBanner();
+  performLogin();
+
+  // Perform logout and verify
+  cy.log("The user tries to logout...");
+  performLogout();
+}
+
+// Validate logout functionality from an article page
+export function validateLogoutFromArticlePage() {
+  cy.log("Verify the logout scenario from the article page");
+
+  // Visit the homepage URL, accept the cookie banner, and perform login
+  cy.visit(url, { timeout: 20000 });
+  cy.acceptCookieBanner();
+  performLogin();
+
+  // Navigate to the first article and perform logout
+  cy.get(articleLink).eq(0).click();
+  performLogout();
+}
+
+// Validate login attempt with invalid credentials
+export function validateLoginForInvalidCreds() {
+  cy.log("Verify when user tries to login using invalid credentials");
+
+  // Visit the homepage URL and accept the cookie banner
+  cy.visit(url, { timeout: 20000 });
+  cy.acceptCookieBanner();
+
+  // Click the login button to navigate to the login page
+  cy.get(loginButton).click();
+
+  // Wait for 2 seconds for the login page to fully load
+  cy.wait(2000);
+
+  // Enter invalid credentials
+  cy.get(creds).eq(0).type("testing@gmail.com");
```

Comment:
All the const values or common values , might be good in fixtures or const files

Reply:
else can create const value on top

Reply:
updated